src/receive_imf.rs

```diff
@@ -1533,6 +1533,40 @@ async fn add_parts(
                 "Edit message: rfc724_mid {rfc724_mid:?} not found."
             );
         }
+    } else if let Some(rfc724_mid_list) = mime_parser.get_header(HeaderDef::ChatDelete) {
+        chat_id = DC_CHAT_ID_TRASH;
+        if let Some(part) = mime_parser.parts.first() {
+            let showpadlock = part
+                .param
+                .get_bool(Param::GuaranteeE2ee)
+                .unwrap_or_default();
+            let mut modified_chat_ids = HashSet::new();
+            let mut msg_ids = Vec::new();
+
+            let rfc724_mid_vec: Vec<&str> = rfc724_mid_list.split_whitespace().collect();
+            for rfc724_mid in rfc724_mid_vec {
+                if let Some((msg_id, _)) = message::rfc724_mid_exists(context, rfc724_mid).await? {
+                    if let Some(msg) = Message::load_from_db_optional(context, msg_id).await? {
+                        if msg.from_id == from_id {
+                            if showpadlock || !msg.get_showpadlock() {
```

Comment:
This line can also be simplified to `if showpadlock {` then

Reply:
i additionally moved the check to the outer `if`